src/Features/Core/Portable/ConvertLinq/AbstractConvertLinqProvider.cs

```diff
@@ -0,0 +1,143 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.Simplification;
+
+namespace Microsoft.CodeAnalysis.ConvertLinq
+{
+    internal abstract class AbstractConvertLinqProvider : CodeRefactoringProvider
+    {
+        public sealed override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var document = context.Document;
+            if (document.Project.Solution.Workspace.Kind == WorkspaceKind.MiscellaneousFiles)
```

Comment:
>if (document.Project.Solution.Workspace.Kind == WorkspaceKind.MiscellaneousFiles) [](start = 12, length = 81)

Why this specific check?  Is there something specific about this provider that it will not work for MiscellaneousFiles workspace?

Do we do it as a defensive check in all IDE fix/refactoring providers?

Reply:
This one exists in multiple refactoring providers. Since @cyrus.najmabadi@gmail.com also does not know what is its purpose, let us remove it.

---
In reply to: [176225931](https://github.com/dotnet/roslyn/pull/25362#discussion_r176225931) [](ancestors = 176225931)

Reply:
Yeah, it's a holdover from ye olde dayz of roslyn.  I don't think it serves much purpose anymore.  As long as the refactoring is not going to be making structural changes to the project, and as long as yuo don't need to query real options, then it's fine to offer.  So basically simplefile changes are pretty much always ok.